src/main/java/Duke.java

```diff
@@ -1,10 +1,226 @@
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static boolean isEmptyDescription(String description) {
+        return ("".equals(description));
+    }
+
+    public static void markAsDone(int taskNum) {
+        if (taskNum <= 0 || taskNum > tasks.size()) {
+            System.out.println("\tInvalid task number!");
+            return;
+        } else if (tasks.get(taskNum - 1).isDone()) {
+            System.out.println("\tTask already done!");
+            return;
+        }
+        tasks.get(taskNum - 1).setDone(true);
+        printHorizontalLine();
+        System.out.println("\t Nice! I've marked this task as done:");
+        System.out.println("\t   " + tasks.get(taskNum - 1));
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static void updateFile() {
+        try {
+            writeTasksToFile();
+        } catch (IOException e) {
+            // relative path is different for testing
+            System.out.println("Check file path!");
+        }
+    }
+
+    public static void writeTasksToFile() throws IOException {
+        FileWriter file = new FileWriter(DIRECTORY_NAME + "/" + FILE_NAME);
+        for (Task t : tasks) {
+            file.write(t.getCode() + "|" + (t.isDone() ? "1" : "0") + "|" +
+                    t.getDescription() + "|" + t.getDatetime() + System.lineSeparator());
+        }
+        file.close();
+    }
+
+    public static void loadTasks() {
+        File directory = new File(DIRECTORY_NAME);
+        boolean directoryCreated = false;
+        try {
+            if (!directory.exists()) {
+                directoryCreated = directory.mkdir();
+            }
+        } catch (Exception e) {
+            e.getStackTrace();
+        }
+        // created = True -> first time running -> no need to load
+        // created = False -> directory exists OR fail to create (exists)
+        if (!directoryCreated) {
+            // handle directory exists case
+            try {
+                readTasksFromFile();
+            } catch (FileNotFoundException e) {
+                // relative path is different for testing
+                System.out.println("File not found!");
+            }
+        }
+    }
+
+    public static void readTasksFromFile() throws FileNotFoundException {
+        File file = new File(DIRECTORY_NAME + "/" + FILE_NAME);
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNext()) {
+            processLine(scanner.nextLine());
+        }
+    }
+
+    public static void processLine(String line) {
+        String[] parts = line.split("\\|");
+        boolean done = (!"0".equals(parts[1]));
+        switch (parts[0]) {
+        case "T":
+            tasks.add(new Todo(done, parts[2]));
+            break;
+        case "E":
+            tasks.add(new Event(done, parts[2], parts[3]));
+            break;
+        case "D":
+            tasks.add(new Deadline(done, parts[2], parts[3]));
+            break;
```

Comment:
Perhaps it might be a good idea to give temporary labels to each of your parts such that you'd grasp how the command string should be formatted.

Reply:
Thanks for catching that.